_posts/2023-17-10-remote-attestation-blog-part1.markdown

```diff
@@ -0,0 +1,137 @@
+
+
+<div style="text-align: center;">
+<img  src="/Users/margieruffin/Desktop/Research/Keylime/keylime.github.io/assets/images/RemoteAttestation.png" alt="Remote Attestation Diagram">
+</div>
+
+**Figure 1: Remote Attestation is used to provide trust between an untrusted party and a trusted party. You can use services like Measured Boot Attestation and Runtime Integrity Monitoring to prove systems are trustworthy.**
+
+In this article, we will detail the different aspects of remote attestation based on the following sections:
+<ul>
+    <li> What is Secure Boot? </li>
+    <li> What is Measured Boot? </li>
+    <li> Measured Boot and Secure Boot go Hand in Hand </li>
+    <li> Integrity Measurement Architecture (IMA) for Continuous Attestation</li>
+    <li> Introduction to Keylime, A Remote Attestation Software </li>
+</ul>
+
+<h3 style="text-align left; color:#6B98BF;"> What is Secure Boot? </h3>
+
+Trusted boot technologies rely on a “Root of Trust” (RoT), a source that can always be trusted within a cryptographic system. The RoT is usually a piece of code or hardware that’s been hardened well enough that it is unlikely to be compromised. The RoT cannot be modified at all or cannot be modified without cryptographic credentials. 
+
+Unified Extended Firmware Interface (UEFI) Secure Boot is a security measure developed to ensure that a device is booted using only software trusted by the Original Equipment Manufacturer (OEM). The goal is to prevent malicious software from being loaded and executed early in the boot process. During secure boot, the kernel checks if a cryptographic signature, a hash value, on the boot loader matches a pre-loaded value signed by the manufacturer, stored inside the BIOS, that anyone can check using a public key made available. 
+
+Secure boot prevents boot if the two signatures, the preloaded value, and the calculated value during the boot process do not match. If the decrypted hash does not match the new one just taken, the secure boot will fail for reasons such as someone changing the kernel. This process of checking the kernel's signatures occurs before the OS ever runs.
+
+<h3 style="text-align left; color:#6B98BF;"> What is Measured Boot? </h3>
+
+Measured Boot uses the same RoT as Secure Boot; however, unlike Secure Boot, it will check each startup component, including firmware, all the way up to the boot drivers. A hash is taken at each boot process step for the next object(s) in the chain. The hashes are stored inside the TPM so that they can be securely retrieved later to find out what objects were encountered. With Measured Boot, the boot process is never stopped, but it provides the necessary information to detect attacks. 
```

Comment:
```
Measured Boot uses the same RoT as Secure Boot
```

This statement is false. The RoT for Secure Boot is the DB in the firmware. The RoT for Measured Boot is the TPM. 

```suggestion
Unlike Secure Boot, Measured Boot will measure each startup component, including firmware, all the way up to the boot drivers. A hash is taken at each boot process step for the next object(s) in the chain. The hashes are extended to the TPM  and added to a event log so that they can be securely retrieved later to find out what objects were encountered. With Measured Boot, the boot process is never stopped, but it provides the necessary information to detect attacks. 
```